File: py/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        t = head
        nthp = head
        i = 0
        prev = head
        while t is not None:
            i += 1
            t = t.next
            if i > n:
                prev = nthp
                nthp = nthp.next
        if nthp != head:
            prev.next = nthp.next
            nthp = None
        else:
            if head.next is not None:
                head = head.next
            else:
                head = None
        return head
            

File: java/19. Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        //int a[];
        int k=0;
        ListNode p=head;
        while(p!=null)
        {
            k++;
            p=p.next;

         }
        int s=k-n+1;
        if(s==1)
        {
            p=head.next;
            
            head=p;
         }
        else{
        p=head;
        ListNode q=head;
        for(int i=1;i<s;i++)
        {
            q=p;
            p=p.next;
        }
        q.next=p.next;
            
        }
       return head;
    }
}